commons/src/test/java/io/aiven/kafka/tiered/storage/commons/storage/filesystem/FileSystemStorageFactoryTest.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2023 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tiered.storage.commons.storage.filesystem;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+
+import io.aiven.kafka.tiered.storage.commons.storage.ObjectStorageFactory;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.NullSource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNoException;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class FileSystemStorageFactoryTest {
+    @ParameterizedTest
+    @NullSource
+    @ValueSource(booleans = { false })
+    void testNotAllowOverwriting(final Boolean overwriteEnabled,
+                                 @TempDir final Path tempDir) throws IOException {
+        final File fsRoot = tempDir.toFile();
+        final ObjectStorageFactory osFactory = new FileSystemStorageFactory();
+        final HashMap<String, String> config = new HashMap<>();
+        config.put("root", fsRoot.toString());
+        if (overwriteEnabled != null) {
```

Comment:
Do we really need the `NullSource` and null check here? I guess the default value will be false anyways and this should not affect anything or am I missing something?

Reply:
> I guess the default value will be false anyways and this should not affect anything or am I missing something?

This is right, but this is tested here. However, I think this property should be tested on the `FileSystemStorageConfig` level, so added a new test class.